### PR TITLE
Feature/verify

### PR DESCRIPTION
**Pull request by MishaFomenko**
## Describe your changes

- added verify email page/form
- added create new password page/form
- added verify otp code page/form

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit f

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

image
image
image
image

### CODE REVIEW

File: src/components/otpForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        {new Array(6).fill(null).map((_, index) => (
+          <Controller
+            key={index}
+            name={`otpInput${index}` as keyof OtpFormValue}
+            control={control}
+            render={({ field }) => (
+              <input
+                type="text"
+                placeholder="1"
+                maxLength={1}
```
Comment (eugeneiv1) [inline src/components/otpForm/index.tsx]: may be values for placeholder and maxLength should be in const?

File: src/pages/verifyOtpPage/index.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import { useTranslation } from 'react-i18next';
+import {
+  FormHeader,
+  LogoContainer,
+  FormSection,
+  SubTitle,
+  Title,
+  FormContainer,
+} from 'components/sharedUI/form/styles';
+import bgImg from 'src/assets/images/sign-in.png';
```
Comment (eugeneiv1) [inline src/pages/verifyOtpPage/index.tsx]: path is ok?

File: src/redux/authApiSlice.ts [hunk 3]
```diff
@@ -1,31 +1,68 @@
...
 import { apiSlice } from './apiSlice';
+import {
+  RequestLogin,
+  RequestRegistration,
+  ResponseLogin,
+  ResponseRegistration,
+  RequestNewPass,
+  ResponseVerifyOtp,
+  RequestVerifyOtp,
+} from './types';
```
Comment (eugeneiv1) [inline src/redux/authApiSlice.ts]: path

File: src/components/otpForm/index.tsx (hunk 1 above)
Comment (MishaFomenko) [reply src/components/otpForm/index.tsx]: I extracted it into variables, but I will keep them at the top of this file

File: src/pages/verifyOtpPage/index.tsx (hunk 2 above)
Comment (MishaFomenko) [reply src/pages/verifyOtpPage/index.tsx]: fixed

File: src/redux/authApiSlice.ts (hunk 3 above)
Comment (MishaFomenko) [reply src/redux/authApiSlice.ts]: fixed

File: src/components/newPassForm/index.tsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        .test(
+          'isRepeatedPasswordValueMatched',
+          t('validation.passwordMatch'),
+          (value, { parent }) => !value || value === parent.password,
+        )
+        .min(passLengthLim, t('validation.password'))
+        .required(t('validation.credentials')),
+    })
+    .required();
+  const {
```
Comment (lyudad) [inline src/components/newPassForm/index.tsx:43]: empty space above

File: src/components/newPassForm/index.tsx [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        type: 'server',
+        message: t('restorePassword.server'),
+      });
+    }
+  };
+
+  const handlePassword = () => {
+    setPasswordShown(!isPasswordShown);
+  };
+
```
Comment (lyudad) [inline src/components/newPassForm/index.tsx]: move all logic to custom hook

File: src/components/otpForm/index.tsx [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+    }
+  };
+
+  useEffect(() => {
+    if (timer > 0) {
+      const timeoutId = setTimeout(() => setTimer(timer - 1), clockPrecision);
+      return () => clearTimeout(timeoutId);
+    }
+  }, [timer]);
+
```
Comment (lyudad) [inline src/components/otpForm/index.tsx]: move all logic above to custom hook

File: src/components/sharedUI/form/styles.ts [hunk 7]
```diff
@@ -0,0 +1,398 @@
...
+
+  @media screen and (max-width: ${({ theme }) => theme.breakpoint.tablet}) {
+    color: ${({ theme }) => theme.colors.white};
+  }
+
+  @media screen and (min-width: ${({ theme }) => theme.breakpoint.desktop}) {
+    color: ${({ theme }) => theme.colors.black};
+  }
+`;
+export const SubTitle = styled.h2<{ theme?: Theme }>`
```
Comment (lyudad) [inline src/components/sharedUI/form/styles.ts:250]: empty line above

File: src/components/sharedUI/form/styles.ts [hunk 8]
```diff
@@ -0,0 +1,398 @@
...
+
+  @media screen and (max-width: ${({ theme }) => theme.breakpoint.tablet}) {
+    color: ${({ theme }) => theme.colors.white};
+  }
+
+  @media screen and (min-width: ${({ theme }) => theme.breakpoint.desktop}) {
+    color: ${({ theme }) => theme.colors.black};
+  }
+`;
+export const FormLink = styled(Link)<{ theme?: Theme }>`
```
Comment (lyudad) [inline src/components/sharedUI/form/styles.ts:266]: empty line above

File: src/pages/newPassPage/index.tsx [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  FormContainer,
+} from 'components/sharedUI/form/styles';
+import bgImg from 'assets/images/new-pass.png';
+import { useAppSelector } from 'redux/hooks.ts';
+import { links } from 'const/links';
+
+const NewPassPage = () => {
+  const { t } = useTranslation();
+  const user = useAppSelector(state => state.auth.user);
+  return (
```
Comment (lyudad) [inline src/pages/newPassPage/index.tsx:22]: empty line above

File: src/pages/verifyEmailPage/index.tsx [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+import bgImg from 'assets/images/verification.png';
+import { ArrowBackLink } from 'components/arrowBackLink';
+import { useAppSelector } from 'redux/hooks.ts';
+import { links } from 'const/links';
+import { otpFormVersions } from 'const/constants';
+
+export const VerifyEmailPage = () => {
+  const { t } = useTranslation();
+  const user = useAppSelector(state => state.auth.user);
+  return (
```
Comment (lyudad) [inline src/pages/verifyEmailPage/index.tsx:21]: empty line above

File: src/pages/verifyOtpPage/index.tsx [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+import bgImg from 'assets/images/sign-in.png';
+import { ArrowBackLink } from 'components/arrowBackLink';
+import { useAppSelector } from 'redux/hooks.ts';
+import { links } from 'const/links';
+import { otpFormVersions } from 'const/constants';
+
+export const VerifyOtpPage = () => {
+  const { t } = useTranslation();
+  const user = useAppSelector(state => state.auth.user);
+  return (
```
Comment (lyudad) [inline src/pages/verifyOtpPage/index.tsx:21]: empty line above

**Review by lyudad — APPROVED** [review]

**Pull request merged**